src/passes/function_passes/Utils/Utils.cpp

```diff
@@ -2,28 +2,17 @@
 
 using namespace llvm;
 
-bool isPatchpoint(Function *fun) {
-  Module *mod = fun->getParent();
-  auto patchpointIntrinsicVoid = Intrinsic::getDeclaration(
-      mod, Intrinsic::experimental_patchpoint_void);
-  auto patchpointIntrinsici64 = Intrinsic::getDeclaration(
-      mod, Intrinsic::experimental_patchpoint_i64);
-  return fun == patchpointIntrinsicVoid || fun == patchpointIntrinsici64;
-}
-
-/*
- * Check if `fun` is `llvm.experimental.stackmap` or
- * `llvm.experimental.patchpoint.*`.
- */
-bool producesStackmapRecords(Function *fun) {
+PatchpointType getPatchpointType(Function *fun) {
   Module *mod = fun->getParent();
   auto stackmapIntrinsic = Intrinsic::getDeclaration(
       mod, Intrinsic::experimental_stackmap);
   auto patchpointIntrinsicVoid = Intrinsic::getDeclaration(
       mod, Intrinsic::experimental_patchpoint_void);
   auto patchpointIntrinsici64 = Intrinsic::getDeclaration(
       mod, Intrinsic::experimental_patchpoint_i64);
-  return fun == stackmapIntrinsic || fun == patchpointIntrinsicVoid
-      || fun == patchpointIntrinsici64;
+  unsigned int type = 0;
+  type |= (fun == stackmapIntrinsic);
```

Comment:
I'd have tried to use the macros you defined here, rather than use magic numbers.